File: Algorithms/src/1653. Minimum Deletions to Make String Balanced/Code/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumDeletions = function (s) {
  let countB = 0;
  let deletions = 0;

  for (let ch of s) {
    if (ch === "b") {
      countB++;
    } else {
      deletions = Math.min(deletions + 1, countB);
    }
  }
  return deletions;
};

File: Algorithms/src/1653. Minimum Deletions to Make String Balanced/Code/solution.go
func minimumDeletions(s string) int {
    countB := 0
    deletions := 0

    for _, ch := range s {
        if ch == 'b' {
            countB++
        } else {
            if deletions+1 < countB {
                deletions = deletions + 1
            } else {
                deletions = countB
            }
        }
    }
    return deletions
}
